PROTO_tests/tests/series_test.py

```diff
@@ -0,0 +1,193 @@
+INTEGRAL_TYPES = [ak.int64, ak.uint64, ak.bool, ak.bigint]
+
+
+class TestSeries:
+    @pytest.mark.parametrize("dtype", DTYPES)
+    def test_series_creation(self, dtype):
+        idx = ak.arange(3, dtype=dtype)
+        for val in idx, ak.array(["A", "B", "C"]):
+            ans = ak.Series(data=val, index=idx).to_list()
+            for series in (
+                ak.Series(data=val, index=idx),
+                ak.Series(data=val),
+                ak.Series(val, idx),
+                ak.Series(val),
+                ak.Series((idx, val)),
+            ):
+                assert isinstance(series, ak.Series)
+                assert isinstance(series.index, ak.Index)
+                assert series.to_list() == ans
+
+        with pytest.raises(TypeError):
+            ak.Series(index=idx)
+
+        with pytest.raises(TypeError):
+            ak.Series((ak.arange(3),))
+
+        with pytest.raises(TypeError):
+            ak.Series()
+
+        with pytest.raises(ValueError):
+            ak.Series(data=ak.arange(3), index=ak.arange(6))
+
+    @pytest.mark.parametrize("dtype", INTEGRAL_TYPES)
+    @pytest.mark.parametrize("dtype_index", [ak.int64, ak.uint64])
+    def test_lookup(self, dtype, dtype_index):
+        pda = ak.arange(3, dtype=dtype)
+        for val in pda, ak.array(["A", "B", "C"]):
+            s = ak.Series(data=val, index=ak.arange(3, dtype=dtype_index))
+
+            for key in (
+                1,
+                ak.Index(ak.array([1], dtype=dtype_index)),
+                ak.Index(ak.array([0, 2], dtype=dtype_index)),
+            ):
+                lk = s.locate(key)
+                assert isinstance(lk, ak.Series)
+                key = ak.array(key) if not isinstance(key, int) else key
+                assert (lk.index == s.index[key]).all()
+                assert (lk.values == s.values[key]).all()
+
+            # testing multi-index lookup
+            mi = ak.MultiIndex([pda, pda[::-1]])
+            s = ak.Series(data=val, index=mi)
+            lk = s.locate(mi[0])
+            assert isinstance(lk, ak.Series)
+            assert lk.index.index == mi[0].index
+            assert lk.values[0] == val[0]
+
+            # ensure error with scalar and multi-index
+            with pytest.raises(TypeError):
+                s.locate(0)
+
+            with pytest.raises(TypeError):
+                s.locate([0, 2])
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", NO_STRING)
+    def test_add(self, prob_size, dtype):
+        added = ak.Series(ak.arange(prob_size // 2, dtype=dtype)).add(
+            ak.Series(
+                data=ak.arange(prob_size // 2, prob_size, dtype=dtype),
+                index=ak.arange(prob_size // 2, prob_size),
+            )
+        )
+        assert added.index.to_list() == ak.arange(prob_size).to_list()
+        assert added.values.to_list() == ak.arange(prob_size, dtype=dtype).to_list()
```

Comment:
since we are parameterizing on `prob_size` this might get too big to call `.to_list`. I would suggest we do this instead
```python
assert (added.index == ak.arange(prob_size)).all()
assert (added.values == ak.arange(prob_size, dtype=dtype)).all()
```

Reply:
Updated